### PR TITLE
fix(Collaboration): Allow to enforce strict email format on sharees API

### PR DESCRIPTION
**Pull request by susnux**
## Summary
Only recommend a email address as sharee if it satisfies the strict format checking. Meaning before `a@b` was valid, while now with strict checking it is invalid and needs to be at least `a@b.c`

### Screenshots
before|after
---|---
Screenshot_20240521_174514|Screenshot_20240521_174502

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by susnux** [comment]
@AndyScherzinger this would fix the email share issue

**Comment by susnux** [comment]
Should we add an UI toggle?

**Comment by AndyScherzinger** [comment]
> Should we add an UI toggle?

Don't think that is needed. A config value is just fine. This is nothing that gets changed, you chose one or the other (forever) I suppose.

File: lib/private/Collaboration/Collaborators/MailPlugin.php [hunk 1]
```diff
@@ -251,15 +254,24 @@ public function search($search, $limit, $offset, ISearchResult $searchResult): b
...
-					'shareWith' => $search,
-				],
-			];
+		if ($search && !$searchResult->hasExactIdMatch($emailType)) {
+			$strictMailCheck = $this->config->getAppValue('core', 'shareapi_enforce_strict_email', 'yes') === 'yes';
+			$validator = new EmailValidator();
+			$validation = match ($strictMailCheck) {
+				false => new RFCValidation(),
+				true => new NoRFCWarningsValidation(),
+			};
```
[the remark this answers is not in the document]
Comment (susnux) [reply lib/private/Collaboration/Collaborators/MailPlugin.php:263]: No strong opinion here :)

**Review by Altahrim — APPROVED** [review]

**Review by come-nc — APPROVED** [review]
A bit weird that the config switch only applies to shareapi and not more globally to email validation.

**Comment by susnux** [comment]
See: https://github.com/nextcloud/server/pull/45570

**Comment by susnux** [comment]
Closed in favor of https://github.com/nextcloud/server/pull/45570

**Pull request closed without merging**